### PR TITLE
Add storage api for new http backend

### PR DESCRIPTION
**Pull request by al8n**

### CODE REVIEW

File: nodes/nomos-node-api/src/http/backend/axum.rs [hunk 1]
```diff
@@ -29,6 +29,7 @@ pub struct AxumBackendSettings {
     pub cl: OverwatchHandle,
     pub carnot: OverwatchHandle,
     pub network: OverwatchHandle,
+    pub storage: OverwatchHandle,
```
[the remark this answers is not in the document]
Comment (danielSanchezQ) [reply nodes/nomos-node-api/src/http/backend/axum.rs:32]: Yep, I was wondering the same. But I have seen this is removed in a different PR?

[the remark this answers is not in the document]
Comment (al8n) [reply nodes/nomos-node-api/src/http/backend/axum.rs:32]: Yeah, in the final PR, there is only one left.

[the remark this answers is not in the document]
Comment (zeegomo) [reply nodes/nomos-node-api/src/http/backend/axum.rs:32]: Would it be hard to not add them here instead?

[the remark this answers is not in the document]
Comment (al8n) [reply nodes/nomos-node-api/src/http/backend/axum.rs:32]: It is not hard, actually. I also considered removing copies in the initial PR (metrics-api), but there are some small refactors on trait design in the final PR, and the `metrics-api` one has already been reviewed. So, I chose to remove these handles in the final PR.

[the remark this answers is not in the document]
Comment (zeegomo) [reply nodes/nomos-node-api/src/http/backend/axum.rs:32]: It's not clear what is the final PR. If you open PRs with a specific sequence in mind please put it in the title (e.g. API refactor [2/5] - storage) or description

[the remark this answers is not in the document]
Comment (al8n) [reply nodes/nomos-node-api/src/http/backend/axum.rs:32]: Sorry for the misleading. The final PR is #490. I will do this next time.

**Review by danielSanchezQ — APPROVED** [review]

**Pull request merged**